osu.Game.Rulesets.Taiko/Mods/TaikoModClassic.cs

```diff
@@ -32,7 +74,65 @@ public void Update(Playfield playfield)
             // Since the time range will depend on a positional value, it is referenced to the x480 pixel space.
             float ratio = drawableTaikoRuleset.DrawHeight / 480;
 
-            drawableTaikoRuleset.TimeRange.Value = (playfield.HitObjectContainer.DrawWidth / ratio) * scroll_rate;
+            double timeRange = (playfield.HitObjectContainer.DrawWidth / ratio) * scroll_rate;
+
+            if (mods != null)
```

Comment:
Why are mods being nullchecked here at all, if `ApplyToDrawableRuleset()` not only doesn't do this, but is also already presumed to have executed by this point? (see `Debug.Assert(drawableTaikoRuleset != null)` assertion at start of method).

You could _expand_ that assertion with this nullcheck, but putting it here is weird.

Reply:
This is because from `DrawableRuleset.cs` it seems like `DrawableRuleset.Mods` could potentially be null, but it has nullable disabled.

The null check is merged with the assertion in bc064c166103cbcf4315ee6d6a5badc18c502a3d and a4d66032e3a74023ea88c6aa164682a819914921, but a potentially more preferrable alternative is to declare `mods` as not nullable and initialize it with an empty list, and do a null coalescing assignment in `ApplyToDrawableRuleset`.

Reply:
On a second thought since classic itself is a mod it's probably safe to assume `mods[]` will never be null here? In which case it's probably safe to declare it as non-nullable

Reply:
Change the declaration to be non-null in c0079df2e82f6d2ba2011a258eb4c7254e8dced0